### PR TITLE
`Paywalls`: initial iOS 15 support

### PR DESCRIPTION
**Pull request by NachoSoto**
`PaywallView` and `PaywallViewController` (along with the extensions) still require iOS 16, but this paves the way to support iOS 15 down the road if we need to.

### Main Changes:

- Changed all `@available` annotations
- Added CI tests
- Replaced `ViewThatFits` with the "biggest" version
- Because of that, `scrollableIfNeeded` is always scrollable
- Double `Regex` implementation (keeping the modern one), so once we drop support for iOS 15 we can just keep the modern one.

### TODO: 
- [x] Add snapshots
- [x] Test paywalls
- [x] Updated APITester to verify this

### CODE REVIEW

File: RevenueCatUI/Templates/OnePackageStandardTemplate.swift [hunk 1]
```diff
@@ -21,9 +21,7 @@ struct OnePackageStandardTemplate: TemplateViewType {
         VStack(spacing: self.configuration.mode.verticalSpacing) {
             self.scrollableContent
                 .scrollableIfNecessary()
-                .scrollContentBackground(.hidden)
```
Comment (joshdholtz) [inline RevenueCatUI/Templates/OnePackageStandardTemplate.swift:24]: Was this not needed anymore?

File: RevenueCatUI/Views/FooterView.swift [hunk 2]
```diff
@@ -71,9 +71,13 @@ private struct RestorePurchasesButton: View {
...
         } label: {
-            ViewThatFits {
+            if #available(iOS 16.0, macOS 13.0, tvOS 16.0, *) {
+                ViewThatFits {
+                    Text("Restore purchases", bundle: .module)
+                    Text("Restore", bundle: .module)
+                }
+            } else {
                 Text("Restore purchases", bundle: .module)
-                Text("Restore", bundle: .module)
```
Comment (joshdholtz) [inline RevenueCatUI/Views/FooterView.swift:76]: Is the "Restore" not needed for iOS 15? 🤔

File: RevenueCatUI/Views/FooterView.swift [hunk 3]
```diff
@@ -101,20 +105,28 @@ private struct LinkButton: View {
...
-                    destination: self.url
-                )
+        if #available(iOS 16.0, macOS 13.0, tvOS 16.0, *) {
+            ViewThatFits {
+                ForEach(self.titles, id: \.self) { title in
+                    self.link(for: title, bundle: bundle)
+                }
             }
+        } else if let first = self.titles.first {
+            self.link(for: first, bundle: bundle)
```
Comment (joshdholtz) [inline RevenueCatUI/Views/FooterView.swift:115]: Why is this only showing one link? 🤔

**Review by joshdholtz — COMMENTED** [review]
Some small questions but I'm really happy that we could drop down to iOS 15 support 😊

File: RevenueCatUI/Views/FooterView.swift (hunk 2 above)
Comment (NachoSoto) [reply RevenueCatUI/Views/FooterView.swift:76]: This is an optimization to make it look better if it doesn't fit. We can't use this in iOS 15 so we gotta rely on `Text` clipping the end of the word if it doesn't fit.

File: RevenueCatUI/Views/FooterView.swift (hunk 3 above)
Comment (NachoSoto) [reply RevenueCatUI/Views/FooterView.swift:115]: It's the same as above, we can't use `ViewThatFits`.

**Comment by NachoSoto** [comment]
> I'm really happy that we could drop down to iOS 15 support 😊

Still gotta try it and make sure that these compromises aren't too much.

File: RevenueCatUI/Templates/OnePackageStandardTemplate.swift (hunk 1 above)
Comment (NachoSoto) [reply RevenueCatUI/Templates/OnePackageStandardTemplate.swift:24]: It's hidden by default in a scroll view (it's not a `List`). This isn't available in iOS 15, so I just removed it.

**Review by joshdholtz — APPROVED** [review]

**Pull request merged**